library/pod-security-policy/allow-privilege-escalation/template.yaml

```
@@ -36,6 +36,42 @@ spec:
                 type: string
   targets:
     - target: admission.k8s.gatekeeper.sh
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: containers
+            expression: 'has(object.spec.containers) ? object.spec.containers : []'
+          - name: initContainers
+            expression: 'has(object.spec.initContainers) ? object.spec.initContainers : []'
+          - name: ephemeralContainers
+            expression: 'has(object.spec.ephemeralContainers) ? object.spec.ephemeralContainers : []'
+          - name: exemptImagePrefixes
+            expression: |
+              !has(variables.params.exemptImages) ? [] :
+                variables.params.exemptImages.filter(image, image.endsWith("*")).map(image, string(image).replace("*", ""))
+          - name: exemptImageExplicit
+            expression: |
+              !has(variables.params.exemptImages) ? [] : 
+                variables.params.exemptImages.filter(image, !image.endsWith("*"))
+          - name: exemptImages
+            expression: |
+              (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                container in variables.exemptImageExplicit ||
```

```suggestion
                container.image in variables.exemptImageExplicit ||
```
`variables.exemptImageExplicit` is a list of images, not containers.

---

ack, fixing and relocating code.